From: =?utf-8?q?Jeremy_B=C3=ADcha?= <jeremy.bicha@canonical.com>
Date: Wed, 9 Aug 2023 15:22:19 -0400
Subject: background: set Ubuntu purple

Ubuntu only has a single purple for the terminal background color
which is a dark background so we can't make the foreground color
different between Light and Dark styles.

Only make this change on Ubuntu and set it universally regardless
of desktop session
---
 src/kgx-terminal.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/kgx-terminal.c b/src/kgx-terminal.c
index 233abb1..f5bb294 100644
--- a/src/kgx-terminal.c
+++ b/src/kgx-terminal.c
@@ -139,6 +139,7 @@ update_terminal_colours (KgxTerminal *self)
   KgxTheme resolved_theme;
   GdkRGBA fg;
   GdkRGBA bg;
+  char *os_name;
 
   // Workings of GDK_RGBA prevent this being static
   GdkRGBA palette[16] = {
@@ -160,6 +161,15 @@ update_terminal_colours (KgxTerminal *self)
     GDK_RGBA ("f6f5f4"), // Bright White
   };
 
+  // Set colors for Ubuntu regardless of dark style
+  os_name = g_get_os_info (G_OS_INFO_KEY_NAME);
+  if (!g_strcmp0(os_name, "Ubuntu")) {
+      fg = (GdkRGBA) { 1.0, 1.0, 1.0, 1.0};
+      bg = (GdkRGBA) { 0.188, 0.039, 0.141, 1.0 };
+      vte_terminal_set_colors (VTE_TERMINAL (self), &fg, &bg, palette, 16);
+      return;
+   }
+
   if (!self->settings) {
     return;
   }
